ledger/src/blocktree.rs

```diff
@@ -359,6 +361,7 @@ impl Blocktree {
         shreds: Vec<Shred>,
         leader_schedule: Option<&Arc<LeaderScheduleCache>>,
     ) -> Result<()> {
+        let _lock = self.insert_shreds_lock.lock().unwrap();
```

Comment:
If I delete this line, does that test fail 100% of the time?  If not, it's a bad test. If so, and the test does anything else, it's too big.  At a glance, it looks too big.

Reply:
You're right, it doesn't fail 100% of the time,  right now it really depends on the scheduling. Any better approaches welcome!